flake.nix

```diff
@@ -1,330 +1,79 @@
-
-    in
-    {
-      inherit nixpkgsFor;
-
-      ply-core = {
-        project = perSystem ply-core.projectFor;
-        flake = perSystem (system: (ply-core.projectFor system).flake { });
-      };
-
-      ply-plutarch = {
-        project = perSystem ply-plutarch.projectFor;
-        flake = perSystem (system: (ply-plutarch.projectFor system).flake { });
-      };
-
-      build-all = perSystem (system:
-        (nixpkgsFor system).runCommand "build-all"
-          (self.ply-core.flake.${system}.packages // self.ply-plutarch.flake.${system}.packages)
-          "touch $out");
-
-      test-core = perSystem (system:
-        let pkgs = nixpkgsFor system;
-        in
-        pkgs.runCommand "test-core"
-          (pkgs.lib.attrsets.getAttrs
-            [ "ply-core:test:ply-core-test" ]
-            self.ply-core.flake.${system}.checks) "touch $out");
-
-      test-plutarch = perSystem (system:
-        let pkgs = nixpkgsFor system;
-        in
-        pkgs.runCommand "test-plutarch"
-          (pkgs.lib.attrsets.getAttrs
-            [ "ply-plutarch:test:ply-plutarch-test" ]
-            self.ply-plutarch.flake.${system}.checks) "touch $out");
-
-      packages = perSystem
-        (system:
-          self.ply-core.flake.${system}.packages //
-          self.ply-plutarch.flake.${system}.packages //
-          { devEnv = mkDevEnv system; }
-        );
-
-      checks = perSystem (system:
-        self.ply-core.flake.${system}.checks // self.ply-plutarch.flake.${system}.checks
-        // { formatting-checks = pre-commit-check-for system; });
-
-      check = perSystem (system:
-        (nixpkgsFor system).runCommand "combined-test"
-          {
-            checksss = builtins.attrValues self.checks.${system}
-              ++ builtins.attrValues self.packages.${system} ++ [
-              self.devShells.${system}.ply-core.inputDerivation
-            ];
-          } ''
-          echo $checksss
-          touch $out
-        '');
-
-      apps = perSystem (system: self.ply-core.flake.${system}.apps // self.ply-plutarch.flake.${system}.apps);
-
-      devShells = perSystem (system: rec {
-        default = devEnv;
-        core = self.ply-core.flake.${system}.devShell;
-        plutarch = self.ply-plutarch.flake.${system}.devShell;
-        devEnv = self.packages.${system}.devEnv;
+                  # Non-Haskell shell tools go here
+                  buildInputs = with pkgs; [
+                    nixpkgs-fmt
+                    fd
+                    git
+                    gnumake
+                  ];
+                  shellHook = pre-commit-check.shellHook +
+                    ''
+                      echo $name
+                    '';
+                };
+
+                inputMap = { "https://input-output-hk.github.io/cardano-haskell-packages" = CHaP; };
+              };
+          })
+        ];
+        pkgs = import nixpkgs { inherit system overlays; inherit (haskellNix) config; };
+        flake = pkgs.ply.flake {
```

Comment:
yeah, i don't think it makes sense to overlay packages here as you're instantly taking out the flake of ply out of pkgs; if you want to keep the overlay version, it would at least make sense to export it as a flake output. (like `plyPkgs = pkgs`))

Reply:
Updated to use the project pkg flake directly.